pkg/k8shandler/cluster.go

```diff
@@ -47,183 +46,150 @@ func (elasticsearchRequest *ElasticsearchRequest) CreateOrUpdateElasticsearchClu
 	}
 	wrongConfig = false
 
-	elasticsearchRequest.getNodes()
+	er.getNodes()
 
 	//clearing transient setting because of a bug in earlier releases which
 	//may leave the shard allocation in an undesirable state
-	if elasticsearchRequest.AnyNodeReady() {
-		if success, err := esClient.ClearTransientShardAllocation(); !success {
-			logrus.Warnf("Unable to clear transient shard allocation for cluster %q in namespace %q: %s", elasticsearchRequest.cluster.Namespace, elasticsearchRequest.cluster.Namespace, err.Error())
-		}
-	}
+	er.ensureNoTransitiveShardAllocations()
 
-	progressUnschedulableNodes(elasticsearchRequest.cluster)
+	if err := er.progressUnschedulableNodes(); err != nil {
+		logrus.Warnf("unable to progress unschedulable nodes for %v in namespace %v", er.cluster.Name, er.cluster.Namespace)
+		return er.UpdateClusterStatus()
+	}
 
```

Comment:
Why does this block throw an error now? Can you elaborate a bit for the reviewer's eyes please.

Reply:
it doesn't throw an error per se, but if we fail to progress unscheduled nodes we should leave this current processing block and retry it again.